### ISSUE TITLE
sc.pp.scale stratified by user-provided category

### ISSUE THREAD

**Issue by vitkl** [scverse/scanpy#2142]
- [x] Additional function parameters / changed functionality / changed defaults?
- [ ] New analysis tool: A simple analysis tool you have been using and are missing in `sc.tools`?
- [ ] New plotting function: A kind of plot you would like to seein `sc.pl`?
- [ ] External tools: Do you know an existing package that should go into `sc.external.*`?
- [ ] Other?

Would be great if `sc.pp.scale` could support scaling within a user-defined category - for example scaling the cells generated using 3' vs 5' 10X technology can be used to account for this source of batch effect. If this is easy to implement of your side, I think this option will be useful for many applications.

**Comment by ivirshup** [comment]
Would this be like:

```python
import anndata as ad, scanpy as sc

def scale_by_batch(adata: ad.AnnData, batch_key: str) -> ad.AnnData:
    return ad.concat(
        {
            k: sc.pp.scale(adata[idx], copy=True)
            for k, idx in adata.obs.groupby(batch_key).indices.items()
        },
        merge="first"
    )
```

?

**Comment by vitkl** [comment]
Thanks for replying. Yes, I think so, but ideally keeping the options of sc.pp.scale:
`zero_center, max_value, layer, obsm`

**Comment by ivirshup** [comment]
I would probably just pass those through with kwargs.

At the moment, I'm not sure I would include this within scanpy, since it's fairly easy for users to do on their own and I'm not sure many people would want to use it.

Using a mask to just scale a subset of the data could fit under a possible `mask` keyword argument.

**Comment by flying-sheep** [comment]
Fixed in #2272
